types/types.go

```diff
@@ -38,29 +64,70 @@ type Block struct {
 	// SealFields       []hexutil.Bytes         `json:"sealFields"` //+ ?
 }
 
+//go:generate gencodec -type Transaction -field-override transactionMarshaling -out gen_transaction_json.go
 type Transaction struct {
-	Accesses             *types.AccessList `json:"accessList,omitempty"`
-	BlockHash            *common.Hash      `json:"blockHash"`
-	BlockNumber          *hexutil.Big      `json:"blockNumber"`
-	ChainID              *hexutil.Big      `json:"chainId,omitempty"`
-	From                 common.Address    `json:"from"`
-	Gas                  hexutil.Uint64    `json:"gas"`
-	GasPrice             *hexutil.Big      `json:"gasPrice"`
-	Hash                 common.Hash       `json:"hash"`
-	Input                hexutil.Bytes     `json:"input"`
-	MaxFeePerGas         *hexutil.Big      `json:"maxFeePerGas,omitempty"`
-	MaxPriorityFeePerGas *hexutil.Big      `json:"maxPriorityFeePerGas,omitempty"`
-	Nonce                hexutil.Uint64    `json:"nonce"`
-	R                    *hexutil.Big      `json:"r"`
-	S                    *hexutil.Big      `json:"s"`
-	To                   *common.Address   `json:"to"`
-	TransactionIndex     *hexutil.Uint64   `json:"transactionIndex"`
-	Type                 hexutil.Uint64    `json:"type"`
-	V                    *hexutil.Big      `json:"v"`
-	Value                *hexutil.Big      `json:"value"`
+	Accesses             types.AccessList `json:"accessList,omitempty"`
+	BlockHash            *common.Hash     `json:"blockHash"`
+	BlockNumber          *big.Int         `json:"blockNumber"`
+	ChainID              *big.Int         `json:"chainId,omitempty"`
+	From                 common.Address   `json:"from"`
+	Gas                  uint64           `json:"gas"                            gencodec:"required"`
+	GasPrice             *big.Int         `json:"gasPrice"`
+	Hash                 common.Hash      `json:"hash"`
+	Input                hexutil.Bytes    `json:"input"                          gencodec:"required"`
```

Comment:
Parity has a creates which is the newly created contract address, EVM space also returns this field.
EVM space also returns status field.
Both are optional